File: 1-TwoSum/1-TwoSum.ts
// Last updated: 6/28/2026, 7:41:03 PM
function twoSum(nums: number[], target: number): number[] {
    const map = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        const element = nums[i];
        const remanent = target - element

        if (map.has(remanent)) {
            return [map.get(remanent), i]
        }

        map.set(element, i)
    }

    return [];
};

File: 1-TwoSum/1-TwoSum.rs
// Last updated: 6/28/2026, 7:41:04 PM
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = HashMap::new();

        for (i, &num) in nums.iter().enumerate() {
            let remanent = target - num;

            if let Some(&index) = map.get(&remanent) {
                return vec![index as i32, i as i32];
            }
            map.insert(num, i);
        }
        vec![]
    }
}